pkg/services/ngalert/notifier/channels/opsgenie.go

```diff
@@ -220,3 +249,22 @@ func (on *OpsgenieNotifier) sendDetails() bool {
 func (on *OpsgenieNotifier) sendTags() bool {
 	return on.SendTagsAs == OpsgenieSendTags || on.SendTagsAs == OpsgenieSendBoth
 }
+
+// ExtractAlertKey gets the alert key based on the label set.
+func ExtractAlertKey(alert *types.Alert) notify.Key {
```

Comment:
I've used notify.ExtractGroupKey for reference, but it's in the other package, so I've put it here. Please let me know if there's a better place to put it.

Reply:
Also the key is a string like `label1=value1 label2=value2 `, this should work for the uniqueness right?